### PR TITLE
Simplify and unify fcntl with FIOCLEX and FIONCLEX

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary
Follow up the change from: https://github.com/apache/incubator-nuttx/pull/5573
- vfs: Forward fcntl(F_SETFD...) to ioctl(FIOCLEX|FIONCLEX...) 
- net: Remove psock_fcntl related code

## Impact
Code refactor only

## Testing
Pass CI

### CODE REVIEW

File: net/netdev/netdev_ioctl.c [hunk 1]
```diff
@@ -1673,13 +1673,56 @@ int psock_vioctl(FAR struct socket *psock, int cmd, va_list ap)
...
+
+          DEBUGASSERT(psock->s_sockif != NULL &&
+                      psock->s_sockif->si_sockcaps != NULL);
+          sockcaps = psock->s_sockif->si_sockcaps(psock);
+
+          if ((sockcaps & SOCKCAP_NONBLOCKING) != 0)
+            {
+               if (nonblock && *nonblock)
+                 {
+                   conn->s_flags |= _SF_NONBLOCK;
```
Comment (pkarashchenko) [inline net/netdev/netdev_ioctl.c:1698]:
```suggestion
                   psock->s_flags |= _SF_NONBLOCK;
```

File: net/netdev/netdev_ioctl.c [hunk 2]
```diff
@@ -1673,13 +1673,56 @@ int psock_vioctl(FAR struct socket *psock, int cmd, va_list ap)
...
+                 }
+               else
+                 {
+                   conn->s_flags &= ~_SF_NONBLOCK;
```
Comment (pkarashchenko) [inline net/netdev/netdev_ioctl.c:1702]:
```suggestion
                   psock->s_flags &= ~_SF_NONBLOCK;
```

File: net/netdev/netdev_ioctl.c (hunk 1 above)
Comment (xiaoxiang781216) [reply net/netdev/netdev_ioctl.c]: Fixed

File: net/netdev/netdev_ioctl.c (hunk 2 above)
Comment (xiaoxiang781216) [reply net/netdev/netdev_ioctl.c]: Fixed

**Review by pkarashchenko — APPROVED** [review]

**Pull request merged**